server/plugins/unifi_api_import/unifi_api_import.py

```
@@ -139,45 +184,95 @@ def resolve_parent_mac(uplink_id):
+        # Resolve VLAN/network for devices if available
+        vlan_id = device.get('vlanId')
+        vlan_name = ''
+        if not vlan_id and 'networkId' in device:
+            network = network_lookup.get(device.get('networkId'), {})
+            vlan_id = network.get("vlanId")
+            vlan_name = network.get("name", "")
+
+        # Default management network fallback for gateway/devices
+        if not vlan_id and network_lookup:
+            # Fallback to the first network (e.g., LAN) if none specified
+            default_net = list(network_lookup.values())[0]
```

Ruff (0.16.1)

[warning] Prefer `next(iter(network_lookup.values()))` over single element slice

Replace with `next(iter(network_lookup.values()))`

(RUF015)